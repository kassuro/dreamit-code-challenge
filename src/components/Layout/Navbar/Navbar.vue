<template>
  <nav class="mb-5 flex justify-center bg-blue-800 p-2">
    <div class="container flex items-center">
      <RouterLink
        :to="{ name: 'home' }"
        class="flex-grow text-5xl font-semibold text-white"
      >
        Totto24
      </RouterLink>

      <a href="#" class="nav-item">Login</a>
      <a href="#" class="nav-item">Register</a>
    </div>
  </nav>
</template>

<script lang="ts" setup>
import { RouterLink } from 'vue-router';
</script>

<script lang="ts">
export default {
  name: 'TottoNavbar',
};
</script>

<style lang="postcss">
.nav-item {
  @apply text-xl text-gray-300;
  @apply p-1 px-3;
}

.nav-item:nth-child(odd) {
  @apply border-l border-gray-500 border-opacity-70;
}
</style>
