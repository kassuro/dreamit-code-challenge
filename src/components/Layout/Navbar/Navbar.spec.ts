import { describe, it, expect } from 'vitest';
import { shallowMount } from '@vue/test-utils';

import Navbar from '@/components/Layout/Navbar/Navbar.vue';

describe('Navbar component', () => {
  it('renders', () => {
    const wrapper = shallowMount(Navbar);

    expect(wrapper.element).toMatchSnapshot();
  });
});
