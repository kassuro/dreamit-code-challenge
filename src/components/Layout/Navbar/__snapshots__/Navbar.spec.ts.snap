// Vitest Snapshot v1

exports[`Navbar component > renders 1`] = `
<nav
  class="mb-5 flex justify-center bg-blue-800 p-2"
>
  <div
    class="container flex items-center"
  >
    <router-link-stub
      ariacurrentvalue="page"
      class="flex-grow text-5xl font-semibold text-white"
      custom="false"
      replace="false"
      to="[object Object]"
    />
    <a
      class="nav-item"
      href="#"
    >
      Login
    </a>
    <a
      class="nav-item"
      href="#"
    >
      Register
    </a>
  </div>
</nav>
`;
